lib/ansible/modules/cloud/vmware/vmware_drs.py

```diff
@@ -0,0 +1,279 @@
+    cluster: "{{ cluster_name }}"
+    validate_certs: no
+    vms:
+        - vm1
+        - vm2
+    drs_rule_name: vm1-vm2-affinity-rule-001
+    enabled: True
+    mandatory: True
+    affinity_rule: True
+
+- name: Create DRS Anti-Affinity Rule for VM-VM
+  vmware_drs:
+    hostname: "{{ esxi }}"
+    username: "{{ esxi_username }}"
+    password: "{{ esxi_password }}"
+    cluster: "{{ cluster_name }}"
+    validate_certs: no
+    vms:
+        - vm1
+        - vm2
+    drs_rule_name: vm1-vm2-affinity-rule-001
+    enabled: True
+    mandatory: True
+    affinity_rule: False
+
+- name: Delete DRS Affinity Rule for VM-VM
+  vmware_drs:
+    hostname: "{{ esxi }}"
+    username: "{{ esxi_username }}"
+    password: "{{ esxi_password }}"
+    cluster: "{{ cluster_name }}"
+    validate_certs: no
+    drs_rule_name: vm1-vm2-affinity-rule-001
+    state: absent
+'''
+
+RETURN = r'''#'''
+
+try:
+    from pyVmomi import vim, vmodl
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import (PyVmomi, vmware_argument_spec, wait_for_task,
+                                         find_vm_by_id, find_cluster_by_name)
+
+
+class VmwareDrs(PyVmomi):
+    def __init__(self, module):
+        super(VmwareDrs, self).__init__(module)
+        self.vm_list = module.params['vms']
+        self.cluster_name = module.params['cluster']
+        self.rule_name = module.params['drs_rule_name']
+        self.enabled = module.params['enabled']
+        self.mandatory = module.params['mandatory']
+        self.affinity_rule = module.params['affinity_rule']
+
+        # Sanity check for cluster
+        self.cluster_obj = find_cluster_by_name(content=self.content,
+                                                cluster_name=self.cluster_name)
+        self.vm_obj_list = []
+        if self.cluster_obj is None:
+            self.module.fail_json(msg="Failed to find the cluster %s" % self.cluster_name)
+        # Sanity check for virtual machines
+        self.get_all_vms_info()
+
+    # Getter
+    def get_all_vms_info(self):
+        """
+        Function to get all VM objects using name from given cluster
+
+        """
+        if not self.vm_list:
+            self.module.fail_json(msg="List of virtual machines can not be empty. "
+                                      "Please specify 'vms' parameter")
+        for vm_name in self.vm_list:
+            vm_obj = find_vm_by_id(content=self.content, vm_id=vm_name,
+                                   vm_id_type='vm_name', cluster=self.cluster_obj)
+            if vm_obj is None:
+                self.module.fail_json(msg="Failed to find the virtual machine %s "
+                                          "in given cluster %s" % (vm_name,
+                                                                   self.cluster_name))
+            self.vm_obj_list.append(vm_obj)
+
+    def get_rules_by_cluster(self, cluster_obj=None):
```

Comment:
get_rules_by_cluster is only used by get_rule_key_by_name… maybe you should merge them.
It would allow you to test the rule name while looping through cluster_obj.configuration.rule.

Reply:
ok.